### PR TITLE
Overview section for REST and a bit of restructuring on sidebar

### PR DESCRIPTION
**Pull request by artemkovalyov**
## Context
My vision on our REST offering overview. I was confused for that Youtuber seeing an empty page)
Happy to hear your feedback and incorporate suggestions.

### CODE REVIEW

**Comment by artemkovalyov** [comment]
I can't invite Marco as he's decided not to add himself to this repo so far.

File: docs/java/features/rest/overview.md [hunk 1]
```diff
@@ -4,7 +4,7 @@ title: Overview
 hide_title: false
 hide_table_of_contents: false
 sidebar_label: Overview
-description: SAP Cloud SDK offers type-safe client generatore for REST (Open API) services
+description: SAP Cloud SDK offers typesafe client generatore for REST (Open API) services
```
Comment (sacnl) [inline docs/java/features/rest/overview.md:7]:
```suggestion
description: SAP Cloud SDK offers typesafe client generators for REST (Open API) services
```

File: docs/java/features/rest/overview.md [hunk 2]
```diff
@@ -14,3 +14,45 @@ keywords:
 - sap cloud sdk
 image:
 ---
+
+## Typesafe client for RESTful services in Java  ##
+Cloud SDK for Java allows to generate a typesafe client using [Open API](https://www.openapis.org/) to conveniently work
```
Comment (sacnl) [inline docs/java/features/rest/overview.md:19]:
```suggestion
SAP Cloud SDK for Java is prepared to generate a typesafe clients using [Open API](https://www.openapis.org/) to conveniently work
```

File: docs/java/features/rest/overview.md [hunk 3]
```diff
@@ -14,3 +14,45 @@ keywords:
...
+with various REST APIs available at [SAP API Hub](https://api.sap.com/)
+
+:::info Open API Generator is not released for public use 
+If you're interested in a typed REST API client for SAP service contact us via [cloudsdk@sap.com](mailto:cloudsdk@sap.com)
```
Comment (sacnl) [inline docs/java/features/rest/overview.md:23]:
```suggestion
If you're interested in a typed REST API client for a specific SAP application or service contact us via [cloudsdk@sap.com](mailto:cloudsdk@sap.com) 
```

File: docs/java/features/rest/overview.md [hunk 4]
```diff
@@ -14,3 +14,45 @@ keywords:
...
+
+:::info Open API Generator is not released for public use 
+If you're interested in a typed REST API client for SAP service contact us via [cloudsdk@sap.com](mailto:cloudsdk@sap.com) 
+:::
+
+### Pre-generated typesafe REST API client ###
+We ship pre-generated typesafe REST API clients as modules in collaboration with popular SAP services available on SAP
+Cloud Platform and beyond.
+
+Depending on the scope modules could be available only internally within SAP or publicly released. [SCP Workflow
```
Comment (sacnl) [inline docs/java/features/rest/overview.md:30]:
```suggestion
Depending on the scope modules could be available only internally within SAP or publicly released. [SAP Cloud Platform Workflow
```

File: docs/java/features/rest/overview.md [hunk 5]
```diff
@@ -14,3 +14,45 @@ keywords:
...
+- We take care of change management by continuously updating, integrating and shipping latest version of services that
+  we release.
+
+## I'm providing a service on SAP Cloud Platform. How can I ship it with Cloud SDK? ##
+Reach out to us via internal communication channels and we'll provide you with information on our contribution models. 
+ 
+## Do you plan to release Open API generator? Like you've done for OData? ##
+Because of less deterministic standard and huge variety of Open API services we decided not to release it as of yet. We
+might reconsider this decision if we see a strong use-case and great value for our customers behind such commitment. The
+value should outweight an effort to provide and support such a feature in general public availability.
```
Comment (sacnl) [inline docs/java/features/rest/overview.md:53]:
```suggestion
value should outweigh an effort to provide and support such a feature in general public availability.
```

**Comment by artemkovalyov** [comment]
I'll fix linebreaks. My editor is a bit too eagerly configured for 120 symbols width.
I also need to respect my own rule of using a grammar checker a bit more thoroughly before pushing it)

**Comment by artemkovalyov** [comment]
The line breaks do not harm the generated pages by the way, but I see an issue with the review.

**Pull request merged**